### ISSUE TITLE
openalpr / tesseract installation difficulty (Mac OS Sierra [10.12.16]/ tesseract version issue)?)

### ISSUE THREAD

**Issue by srlarsen1** [openalpr/openalpr#763]
I'm having difficulty getting openalpr installed/ to run.  The error message I'm getting after "installation" is below.  I'm stuck back on Mac OS Sierra (10.12.6 - for other software reasons) and I suspect that's causing problems although my other thought is that it's a version issue with tesseract as my installed files have (ver)4 instead of the 3 mentioned in the error message.

s-macbook-pro:tesseract Scott$ alpr 5497.jpg
dyld: Library not loaded: @loader_path/liblept.3.dylib
  Referenced from: /usr/local/opt/tesseract/lib/libtesseract.3.dylib
  Reason: image not found
Abort trap: 6

I've tried installing openalpr each of the various ways with minor glitches (pasted below) each way.

Error: python is already installed from !
Please `brew uninstall python` first."

fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[2]: *** [openalpr/CMakeFiles/openalpr.dir/ocr/tesseract_ocr.cpp.o] Error 1
make[1]: *** [openalpr/CMakeFiles/openalpr.dir/all] Error 2
make: *** [all] Error 2

Error: Could not link:
/usr/local/share/man/man1/brew.1

Please delete these paths and run `brew update`.
Error: Could not link:
/usr/local/share/doc/homebrew

I've followed all of the recommended instructions from error messages and googled and followed all the articles with similar error messages (this seems to be a problem across packages) instructions with no luck.  That's required installing and reinstalling opencv, tesseract(4.0.0), leptonica, (home)brew, xcode(9.2), python(3.7.1 [2.7.10_2 also installed]), MacPorts (could never get that to work)....  Please help this relative newbie with as basic instructions as you can muster.

Thanks in advance,
Scott

Error while following these instructions - https://github.com/tesseract-ocr/tesseract/issues/1453 :

....
make[2]: *** [libtesseract.la] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
s-macbook-pro:tesseract Scott$ sudo update_dyld_shared_cache
Password:
s-macbook-pro:tesseract Scott$ make training
Need to reconfigure project, so there are no errors

s-macbook-pro:tesseract Scott$ tesseract --version
tesseract 4.0.0
 leptonica-1.77.0
  libgif 5.1.4 : libjpeg 9c : libpng 1.6.36 : libtiff 4.0.10 : zlib 1.2.8 : libwebp 1.0.1 : libopenjp2 2.3.0
 Found AVX
 Found SSE

[Full Terminal Session.txt](https://github.com/openalpr/openalpr/files/2717371/Full.Terminal.Session.txt)

**Comment by VictorPapyshev** [comment]
I am experiencing the same issue on 10.14 Mojave. When installing with Homebrew and running alpr, I get:

Victors-MBP:~ victorpapyshev$ alpr
dyld: Library not loaded: /usr/local/opt/tesseract/lib/libtesseract.3.dylib
  Referenced from: /usr/local/bin/alpr
  Reason: image not found
Abort trap: 6

**Comment by IvanLjubicic** [comment]
Hi. I have the same issue as @VictorPapyshev 
Any news regarding that problem?

**Comment by 2006pmach** [comment]
Same here.

**Comment by srlarsen1** [comment]
I finally got OpenALPR to work.  I did a ton of other stuff - including nuking my Python install - but I suspect what fixed this particular error message was issuing this command in terminal.

`ln -s /usr/local/opt/tesseract/lib/libtesseract.4.dylib /usr/local/opt/tesseract/lib/libtesseract.3.dylib`

It's a single-line command in case it gets wrapped here so be sure to copy it elsewhere in a single line before running it in terminal.  You also may need to change the folder structure if it doesn't match yours.  As far as I can tell it is looking for an older version of a tesseract file and just links it to the newer version.  I got the idea from https://github.com/openalpr/openalpr/issues/381#issuecomment-235260085 so thanks, @kondratk.

**Comment by 2006pmach** [comment]
Yeah I found that out too, but it just lead to the next issue. I am getting now:

```
Error opening data file /usr/local/Cellar/openalpr/2.3.0_1/share/openalpr/runtime_data/ocr/lus.traineddata
Please make sure the TESSDATA_PREFIX environment variable is set to your "tessdata" directory.
Failed loading language 'lus'
Tesseract couldn't load any languages!
No license plates found.
```
Did you get that as well?

**Comment by srlarsen1** [comment]
@2006pmach  I did.  I think I just searched my system for the "lus.traineddata" file, was able to find it and made a copy in the folder in which it was searching.  I think that came after trying to download and install the "eng.traineddata" (not sure what the "lus" is referring to, anyway) so it might have come through that process.  If you can't locate a copy of the file let me know.

**Comment by 2006pmach** [comment]
@srlarsen1 Now I don't get any errors any more. Downloaded the "lus.traineddata" from https://github.com/openalpr/openalpr/tree/master/runtime_data/ocr/tessdata
As far as I understand it contains some model weights etc. obtained by training the character recognition system using US plates. The data for other countries can be downloaded there as well.
